### ISSUE TITLE
Internal error with float increment via typecast

### ISSUE THREAD

**Issue by markjreed** [irmen/prog8#90]
Not sure exactly what the conditions are that trigger this, but this code bombs the compiler during codegen with an internal error:

```
%zeropage basicsafe
%import textio
%import floats

main {
    ubyte[3] left   = [ 1, 2, 3 ]
    ubyte[3] right = [ 2, 1, 3]
    sub score(ubyte l, ubyte r) -> ubyte {
       return (r - l + 1) % 3 * 3 + r
    }
    sub start() {
      ubyte i
      ubyte l
      ubyte r
      float total = 0
      for i in 0 to 2 {
        l = left[i]
        r = right[i]
        total += score(l,r) as float
      }
      floats.print_f(total)
      txt.nl()
    }
}
```

If I change the `total += score(l,r) as float` line to two lines with an intermediate var, it compiles successfully:

```
    result = score(l,r) as float
    total += score
```

However, this then results in an ?OVERFLOW ERROR at runtime on the Commander X16.

**Comment by irmen** [comment]
The OVERFLOW error was fixed by correcting several ROM routine addresses in f2273c0acc2cb98893d92a104ac9016eaf1a919b

However the program below is an adaptation from the one above and shows several more FP issues (on 6502 targets, Virtual target works fine):

```
%zeropage dontuse
%import textio
%import floats
%option no_sysinit

main {
    sub score() -> ubyte {
        cx16.r15++
        return 5
    }

    sub start() {
        float total = 0
        ubyte bb = 5

        ; TODO FIX COMPILER CRASH (6502 target):
        cx16.r0 = 0
        total += cx16.r0 as float

        ; TODO FIX COMPILER CRASH (6502 target):
        total += score() as float

        ; TODO FIX WRONG RESULT (6502 target):
        uword ww = 5
        total += ww as float

        ; TODO FIX WRONG RESULT (6502 target):
        total += bb as float

        ; CORRECT RESULT (5):
        float result = score() as float
        total += result

        floats.print_f(total)
        txt.nl()
    }
}```

**Comment by irmen** [comment]
internal exception should be fixed now in a82b2da16e1e6e0c76180c3fbf387630f4316f6f
